server/cache.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+import "sync"
+
+type cache struct {
+	byName map[string]string
```

Comment:
I tried to use sync.Map but I don't know why didn't Load the cached result so I finally went with this solution.